internal/server/router/meter_query.go

```diff
@@ -160,9 +213,18 @@ func (resp QueryMeterResponse) Render(_ http.ResponseWriter, _ *http.Request) er
 }
 
 // RenderCSV renders the response as CSV.
-func (resp QueryMeterResponse) RenderCSV(w http.ResponseWriter, r *http.Request, groupByKeys []string, meterIDOrSlug string) {
+func (resp QueryMeterResponse) RenderCSV(w http.ResponseWriter, r *http.Request, _groupByKeys []string, meterIDOrSlug string) {
 	records := [][]string{}
 
+	// Filter out the subject from the group by keys
+	groupByKeys := []string{}
```

Comment:
```suggestion
	dataGroupByKeys := make([]string, 0, len(groupByKeys))
```

The slice will have at most the same length as `groupByKeys`